### PR TITLE
Add a basic log formatter for Chromium JSON format.

### PR DESCRIPTION
**Pull request by LukeZielinski**
This is in preparation for integrating WPT into Chromium CI. The output format of build steps in Chromium is defined by the [JSON Test Result format](https://chromium.googlesource.com/chromium/src/+/master/docs/testing/json_test_results_format.md).

This change adds the basics of the formatter, covering the required fields. Future changes may add additional output. Also added unit tests for this formatter.

### CODE REVIEW

File: tools/wptrunner/wptrunner/formatters/tests/test_chromium.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    # Set up the handler.
+    output = StringIO()
+    logger = structuredlog.StructuredLogger("test_a")
+    logger.add_handler(handlers.StreamHandler(output, chromium.ChromiumFormatter()))
+
+    # output a bunch of stuff
+    logger.suite_start(["test-id-1"], run_info={}, time=123)
+    logger.test_start("test-id-1")
+    time.sleep(0.125)
```
Comment (gsnedders) [inline tools/wptrunner/wptrunner/formatters/tests/test_chromium.py]: Is there any need for this at all?

Comment (LukeZielinski) [reply tools/wptrunner/wptrunner/formatters/tests/test_chromium.py]: Oops, lazy c/p by me. Removed.

File: tools/wptrunner/wptrunner/formatters/chromium.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        name_parts = filter(None, name.split("/"))
+        cur_dict = self.tests
+        for name_part in name_parts:
+            if name_part not in cur_dict:
+                cur_dict[name_part] = {}
+            cur_dict = cur_dict[name_part]
+        cur_dict["actual"] = actual
+        cur_dict["expected"] = expected
+
+    def _map_status_name(self, status):
```
Comment (gsnedders) [inline tools/wptrunner/wptrunner/formatters/chromium.py]: Is there any reason why this is a function and not just a dict? I don't have an overly strong preference here.

If you keep it as a function can you add a failure case that throws if the status is unknown?

File: tools/wptrunner/wptrunner/formatters/chromium.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        self.tests = {}
+
+    def _store_test_result(self, name, actual, expected):
+        """
+        Stores the result of a single test in |self.tests|
+        :param str name: name of the test.
+        :param str actual: actual status of the test.
+        :param str expected: expected status of the test.
+        """
+        name_parts = filter(None, name.split("/"))
```
Comment (gsnedders) [inline tools/wptrunner/wptrunner/formatters/chromium.py]: Can you add a comment saying what `None` does as the first arg? It's a bit non-obvious!

Also: why is this needed? Is there any case when we're returning something with `//` in it?

File: tools/wptrunner/wptrunner/formatters/chromium.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        cur_dict = self.tests
+        for name_part in name_parts:
+            if name_part not in cur_dict:
+                cur_dict[name_part] = {}
+            cur_dict = cur_dict[name_part]
```
Comment (gsnedders) [inline tools/wptrunner/wptrunner/formatters/chromium.py]: This whole body of the `for` is equivalent to `cur_dict = cur_dict.setdefault(name_part, {})`.

File: tools/wptrunner/wptrunner/formatters/chromium.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        self.start_timestamp_seconds = data["time"] if "time" in data else time.time()
+
+    def test_end(self, data):
+        actual_status = self._map_status_name(data["status"])
+        expected_status = data["expected"] if "expected" in data else "PASS"
+        self._store_test_result(data["test"], actual_status, expected_status)
+
+        # Update the count of how many tests ran with each status.
+        try:
+            self.num_failures_by_status[actual_status] += 1
```
Comment (gsnedders) [inline tools/wptrunner/wptrunner/formatters/chromium.py]: Can we just use a `collections.defaultdict` here instead?

File: tools/wptrunner/wptrunner/formatters/chromium.py (hunk 3 above)
Comment (LukeZielinski) [reply tools/wptrunner/wptrunner/formatters/chromium.py]: Done - it's to deal with test names having a leading / which produces an empty str in the list from `split`

File: tools/wptrunner/wptrunner/formatters/chromium.py (hunk 2 above)
Comment (LukeZielinski) [reply tools/wptrunner/wptrunner/formatters/chromium.py]: Added a default result, which I chose to just return the status as-is. Not sure which would be better quite yet, this may change after this logger actually gets used for a bit.

File: tools/wptrunner/wptrunner/formatters/chromium.py (hunk 4 above)
Comment (LukeZielinski) [reply tools/wptrunner/wptrunner/formatters/chromium.py]: Done, thanks!

File: tools/wptrunner/wptrunner/formatters/chromium.py (hunk 5 above)
Comment (LukeZielinski) [reply tools/wptrunner/wptrunner/formatters/chromium.py]: Yep, done.

**Review by LukeZielinski — COMMENTED** [review]
Thanks Sam, appreciate the py wisdom.

**Review by gsnedders — APPROVED** [review]
